File: CSharp/Strategy.Duck/ModelDuck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DesignPatterns.Strategy
{
    public class ModelDuck : Duck
    {
        public ModelDuck()
        {
            flyBehaviour = new FlyNoWay();            
            quackBehaviour = new Quack();
        }

        public override void display()
        {
            Console.Out.WriteLine("I am a model Duck");
        }
    }
}

File: Cpp/hfdp/src/strategy/ducks/ModelDuck.cpp
//
// Created by Johnson, Chase on 12/4/16.
//

#include <iostream>
#include "ModelDuck.h"
#include "Quack.h"
#include "FlyNoWay.h"


ModelDuck::ModelDuck() {
    this->setFlyBehaviour(new FlyNoWay());
    this->setQuackBehaviour(new Quack());
}

void ModelDuck::display() {
    std::cout << "I'm a model duck" << std::endl;
}
